pex/dist_metadata.py

```diff
@@ -789,14 +789,20 @@ def as_constraint(self):
 class DistMetadata(object):
     @classmethod
     def from_metadata_files(cls, metadata_files):
```

Comment:
You might get an even more informative message y adding a new `source=""` argument that `DistMetadata.load(...)` below could pass. If that were either the full location or even just the basename, that would reveal the exact artifact with the issue, the sdist or one of the wheels. In the insane world of Python packaging the metadata is not required to be uniform amongst these various artifacts all in the same release version.